athena-dynamodb/src/main/java/com/amazonaws/athena/connectors/dynamodb/util/DDBTypeUtils.java

```diff
@@ -86,66 +96,85 @@ private DDBTypeUtils() {}
      * @param value the value of the field
      * @return the inferred Arrow field
      */
-    public static Field inferArrowField(String key, Object value)
+    public static Field inferArrowField(String key, AttributeValue value)
     {
         logger.debug("inferArrowField invoked for key {} of class {}", key,
-                value != null ? value.getClass() : null);
-        if (value == null) {
+                value != null ? value.toString() : null);
+
+        EnhancedAttributeValue enhancedAttributeValue = EnhancedAttributeValue.fromAttributeValue(value);
+        if (enhancedAttributeValue == null || enhancedAttributeValue.isNull()) {
             return null;
         }
 
-        if (value instanceof String) {
+        if (enhancedAttributeValue.isString()) {
             return new Field(key, FieldType.nullable(Types.MinorType.VARCHAR.getType()), null);
         }
-        else if (value instanceof byte[]) {
+        else if (enhancedAttributeValue.isBytes()) {
             return new Field(key, FieldType.nullable(Types.MinorType.VARBINARY.getType()), null);
         }
-        else if (value instanceof Boolean) {
+        else if (enhancedAttributeValue.isBoolean()) {
             return new Field(key, FieldType.nullable(Types.MinorType.BIT.getType()), null);
         }
-        else if (value instanceof BigDecimal) {
+        else if (enhancedAttributeValue.isNumber()) {
             return new Field(key, FieldType.nullable(new ArrowType.Decimal(38, 9)), null);
         }
-        else if (value instanceof List || value instanceof Set) {
+        else if (enhancedAttributeValue.isSetOfBytes() || enhancedAttributeValue.isSetOfNumbers() || enhancedAttributeValue.isSetOfStrings()) {
```

Comment:
Should we separate isSetOfNumbers?. It seems we are parsing it as`VARBINARY` now and `VARBINARY` as varchar?

Reply:
There is a typo; the check should have been negated;

          ```
  if (!enhancedAttributeValue.isSetOfBytes()) {
                child = new Field(key, FieldType.nullable(Types.MinorType.VARCHAR.getType()), null);
            }
```

Will fix and separate them out for easier read.